### PR TITLE
rgw: remove quota soft threshold

### PR DESCRIPTION
**Pull request by pleiadesian**
According to [this thread in dev@ceph.io](https://lists.ceph.io/hyperkitty/list/dev@ceph.io/thread/QHALPTYRZA2OKLHYNOMY7EZYRXPMXVYU/), the quota soft threshold is not reasonable anymore after dynamic resharding was introduced, which causes expensive checks for sharded buckets. This pull request reverts the 14eabd4aa7b8a2e2c0c43fe7f877ed2171277526 commit to remove the quota soft threshold.

Fixes: 14eabd4aa7b8a2e2c0c43fe7f877ed2171277526
https://tracker.ceph.com/issues/50975
Signed-off-by: Zulai Wang <wangzl31@outlook.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_quota.cc [hunk 1]
```diff
@@ -196,7 +165,9 @@ void RGWQuotaCache<T>::set_stats(const rgw_user& user, const rgw_bucket& bucket,
 }
 
 template<class T>
-int RGWQuotaCache<T>::get_stats(const rgw_user& user, const rgw_bucket& bucket, RGWStorageStats& stats, RGWQuotaInfo& quota, optional_yield y, const DoutPrefixProvider *dpp) {
+int
```
Comment (ofriedma) [inline src/rgw/rgw_quota.cc:168]: I think you added new line by mistake, other than that LGTM

**Review by ofriedma — CHANGES_REQUESTED** [review]

File: src/rgw/rgw_quota.cc (hunk 1 above)
Comment (pleiadesian) [reply src/rgw/rgw_quota.cc]: Reformatted.

Comment (ofriedma) [reply src/rgw/rgw_quota.cc]: Can you please squash the commits into a single commit?

Comment (pleiadesian) [reply src/rgw/rgw_quota.cc]: Done

**Review by ofriedma — APPROVED** [review]
lgtm

**Comment by ofriedma** [comment]
https://pulpito.ceph.com/dang-2021-05-04_12:57:54-rgw-wip-dang-zipper-link-distro-basic-smithi/

**Pull request merged**